### PR TITLE
Fixes for the security audit

### PR DESCRIPTION
**Pull request by luckychess**
Update covers:
- Dockerfile runs `sscd` from unprivileged user
- Bump several go.mod dependencies

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,25 +1,25 @@
 module github.com/sagaxyz/ssc
 
-go 1.22.2
+go 1.22.7
```
Comment (emanuelconunaemme) [inline go.mod:3]: This should be consistent with the dockerfile. Why having two different ones?

File: Dockerfile [hunk 2]
```diff
@@ -18,6 +18,10 @@ COPY --from=build-env /root/build/sscd /usr/bin/
 
 RUN apk add gcompat bash curl
 
+RUN addgroup -S sscd && adduser -S sscd -G sscd
+USER sscd
+WORKDIR /home/sscd
```
Comment (emanuelconunaemme) [inline Dockerfile:23]: Wait, this is a breaking change and will require a migration. Is it worth? If so, we will need to update the guides and everywhere we are using docker. Basically, current validators running as root have all the files under `/root/.ssc`. If you change the home that directory will be empty.

**Comment by emanuelconunaemme** [comment]
I realized after that this was closed. Just making sure you are not merging the same changes elsewhere.

File: Dockerfile (hunk 2 above)
Comment (luckychess) [reply Dockerfile:23]: Yep, it's not worth to do now so I closed this PR, we have more than enough to do without it.

**Pull request closed without merging**